app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V1
+      class UsersController < ApplicationController
+        skip_before_action :doorkeeper_authorize!, only: %i[create]
+    
+        def create
+          user = User.new(email: user_params[:email], password: user_params[:password])
+    
+          client_app = Doorkeeper::Application.find_by(uid: params[:client_id])
+    
+          return render(json: { error: 'Invalid client ID'}, status: 403) unless client_app
+    
+          if user.save
+            # create access token for the user, so the user won't need to login again after registration
+            access_token = Doorkeeper::AccessToken.create(
+              resource_owner_id: user.id,
+              application_id: client_app.id,
+              refresh_token: generate_refresh_token,
+              expires_in: Doorkeeper.configuration.access_token_expires_in.to_i,
+              scopes: ''
+            )
+            
+            # return json containing access token and refresh token
+            # so that user won't need to call login API right after registration
+            render(json: {
+              user: {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/blyscuit/rails-example-one/blob/ffe479ef97670c7333f37452d55369f226bbc764/app/controllers/api/v1/users_controller.rb#L26"></span>Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.</td>
    </tr>
  </tbody>
</table>